packages/vue-client/src/components/GameCreation/GameCreationForm.vue

```diff
@@ -33,14 +33,18 @@ const createGame = async () => {
   if (
     !config ||
     typeof config !== "object" ||
-    (config.time_control !== undefined &&
-      typeof config.time_control.mainTimeMS !== "number")
+    (timeControlConfig !== null &&
+      typeof timeControlConfig.mainTimeMS !== "number")
   ) {
     console.log("game creation form is invalid");
     return;
   }
   //ToDo: add form validation
 
+  if (timeControlConfig !== null) {
+    config.time_control = timeControlConfig;
+  }
```

Comment:
Nit: I don't love mutating `config` - if for some reason we stayed on the page instead of redirecting (maybe the `POST` fails?).  The time control will remain on `config`.

Then, if we set the config to "Unlimited" again, the wrong time control will be used.

---

An alternative is to leave config unmutated, but pass `{ ...config, time_control: timeControlConfig }` to the `POST` call (`...` is the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals))

Reply:
wait, why was that nonsense? I think your suggestion sounded good:

pass { ...config, time_control: timeControlConfig } to the POST call

Reply:
Hmm I guess immutability is still not bad idea, but I saw the other solution, and didn't want to be an FP zealot 😅 

Also there is a slight difference between

```
{ ...config, time_control: timeControlConfig }
```

and 

```
delete config["time_control"]
```

because the former will send a config like:

```
{ width: 19, height: 19, time_control: null }
```

And I'm not sure if that matters.

Reply:
I see, but you make a good point. I've added your suggestion together with a null check, and moved the type check function to shared (because it was really used on both ends).

In retrospective, I think I should have avoided working so much with undefined when I started, just setting it to null is simpler as opposed to not setting the property at all.

I'm learning ^^

Reply:
> didn't want to be an FP zealot 😅

by the way, I have no idea what that means xD

Reply:
I'd guess FP means Functional Programming, but I might be totally wrong.

Reply:
Yep, [functional programming](https://en.wikipedia.org/wiki/Functional_programming)!  (all variables are immutable in FP)